src/components/Header/Header.tsx

```diff
@@ -1,156 +1,173 @@
 import * as React from 'react';
-import { ColorPaletteProp } from '@mui/joy/styles';
-import Avatar from '@mui/joy/Avatar';
-import Badge from '@mui/joy/Badge';
-import Box from '@mui/joy/Box';
-import Button from '@mui/joy/Button';
-import Input from '@mui/joy/Input';
-import IconButton from '@mui/joy/IconButton';
-import ListDivider from '@mui/joy/ListDivider';
-import ListItemDecorator from '@mui/joy/ListItemDecorator';
-import Menu from '@mui/joy/Menu';
-import MenuItem from '@mui/joy/MenuItem';
-import Typography from '@mui/joy/Typography';
-import Sheet from '@mui/joy/Sheet';
-import Chip from '@mui/joy/Chip';
-import AddIcon from '@mui/icons-material/Add';
-import BubbleChartIcon from '@mui/icons-material/BubbleChart';
+import Toolbar from '@mui/material/Toolbar';
+import MenuItem from '@mui/material/MenuItem';
+import Menu from '@mui/material/Menu';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import FormGroup from '@mui/material/FormGroup';
+import SearchIcon from '@mui/icons-material/Search';
+import AccountCircle from '@mui/icons-material/AccountCircle';
+import MailIcon from '@mui/icons-material/Mail';
 import NotificationsIcon from '@mui/icons-material/Notifications';
-import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import MoreIcon from '@mui/icons-material/MoreVert';
+import Badge from '@mui/material/Badge';
+import {
+  Search,
+  SearchIconWrapper,
+  IconButtonNew,
+  IconButtonNewOne,
+  IconButtonNewTwo,
+  BoxNew,
+  AppBarNew,
+  TypographyNew,
+  StyledInputBaseNew,
+  BoxNewOne,
+  IconButtonFour,
+  IconButtonFive,
+  IconButtonSix,
+  BoxNewTwo,
+  IconButtonSeven,
+  SwitchNew,
+} from './Header.style';
 
-export default function ColorInversionHeader() {
-  const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);
-  const [color, setColor] = React.useState<ColorPaletteProp>('primary');
-  return (
-    <Sheet
-      variant="solid"
-      color={color}
-      invertedColors
-      sx={{
-        display: 'flex',
-        alignItems: 'center',
-        flexGrow: 1,
-        p: 2,
-        mx: -3,
-        my: -3,
-        borderRadius: { xs: 0, sm: 'xs' },
-        minWidth: 'min-content',
-        ...(color !== 'warning' && {
-          background: (theme) => `linear-gradient(to top, ${theme.vars.palette[color][600]}, ${theme.vars.palette[color][500]})`,
-        }),
+export default function PrimarySearchAppBar() {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [auth, setAuth] = React.useState(true);
```

Comment:
As we spoke in the office, boolean value type should contain **is**  prefix

Reply:
+ add typing for useState

Reply:
add correction